src/components/Modal.tsx

```diff
@@ -11,8 +11,15 @@ const Panel: FC<any> = (props) => (
     leaveFrom="opacity-100"
     leaveTo="opacity-0"
   >
-    <Dialog.Panel {...props}>{props.children}</Dialog.Panel>
+    <Dialog.Panel {...props} />
   </Transition.Child>
 );
 
-export default Panel as typeof Dialog.Panel;
+const TitleFn: FC<any> = (props) => <Dialog.Title {...props} />;
+
+const ModalFnRoot = ModalFn as typeof Dialog.Panel;
+const Title = TitleFn as typeof Dialog.Title;
+
+const Modal = Object.assign(ModalFnRoot, { Title });
```

Comment:
Add `Modal.Title` as a wrapper around `Dialog.Title` simply for consistency, i.e. it is less confusing to see:
```tsx
import Modal from "components/Modal";
import { Dialog } from "@headlessui/react";
...
<Modal>
  <Modal.Title>...</Modal.Title>
  ...
</Modal>
```
then seeing:
```tsx
import Modal from "components/Modal";
...
<Modal>
  <Dialog.Title>...</Dialog.Title>
  ...
</Modal>
```
Also one less import.

Reply:
Based off of official @headlessui implementation of same components we're wrapping
For their implementation, see: https://github.com/tailwindlabs/headlessui/blob/main/packages/%40headlessui-react/src/components/dialog/dialog.tsx#L643